zebrad/src/components/mempool.rs

```diff
@@ -146,6 +166,9 @@ impl Service<Request> for Mempool {
 
     #[instrument(name = "mempool", skip(self, req))]
     fn call(&mut self, req: Request) -> Self::Future {
+        if !self.enabled {
+            return async move { Err(MempoolError::Disabled.into()) }.boxed();
+        }
```

Comment:
Likely Bug: this code will disconnect peers when the mempool is disabled.

Errors are only used when the peer has done something wrong or unexpected.

If we return an error to the inbound service, it gets sent to Zebra's peer connection handler, which disconnects from the peer. (Because peer errors make the connection unrecoverable.) And that can slow down syncing.

But in this case, the peer has done nothing wrong. So we don't want to return an error.
(Similarly, the `Queued` response can contain inner errors, but they are ignored by the inbound service, because the peer has done nothing wrong.)

Along with this fix, can you also:
* document this error handling for the `mempool` and `inbound` service - 
* test that a new `Queue` is handled correctly when the mempool is disabled

Reply:
It might be helpful to split mempool requests into two different categories:
* read-only requests, where the existing code should just work when the mempool is disabled
* write requests, which need to be specifically disabled when the mempool is disabled

But if we're only ever going to have one read-write request, then maybe a comment is enough?

Reply:
Good catch!

The logic and tests were changed in 5244db34 along with the refactoring.

Documentation was changed in 976c19ead - I was not sure were to put it.

Reply:
> Documentation was changed in [976c19e](https://github.com/ZcashFoundation/zebra/commit/976c19ead5ddbb03e16506b82508143d684ddeaf) - I was not sure were to put it.

Looks fine where it is - I just want to help us find these kinds of issues more easily next time.
(So I don't have to read all the way through to zebra-network to check how it works.)